zerver/lib/reminders.py

```diff
@@ -0,0 +1,66 @@
+from django.conf import settings
+from django.utils.translation import gettext as _
+
+from zerver.actions.message_send import internal_send_private_message
+from zerver.lib.exceptions import JsonableError
+from zerver.lib.markdown.fenced_code import get_unused_fence
+from zerver.lib.mention import silent_mention_syntax_for_user
+from zerver.lib.message import access_message
+from zerver.lib.message_cache import MessageDict
+from zerver.lib.url_encoding import near_message_url, topic_narrow_url
+from zerver.models import Message, Stream, UserProfile
+from zerver.models.users import get_system_bot
+
+
+def get_reminder_formatted_content(
+    message: Message, current_user: UserProfile, stream: Stream | None
+) -> str:
+    if message.is_stream_message():
+        assert stream is not None
+        narrow_link = topic_narrow_url(
+            realm=current_user.realm,
+            stream=stream,
+            topic_name=message.topic_name(),
+        )
+        content = _(
+            "You requested a reminder for the following message sent to [{stream_name} > {topic_name}]({narrow_link})."
+        ).format(
+            stream_name=stream.name,
+            topic_name=message.topic_name(),
+            narrow_link=narrow_link,
+        )
+    else:
+        content = _("You requested a reminder for the following private message.")
+
+    # Format the message content as a quote.
+    content += "\n\n"
+    content += _("{user_silent_mention} [said]({conversation_url}):").format(
+        user_silent_mention=silent_mention_syntax_for_user(message.sender),
+        conversation_url=near_message_url(current_user.realm, MessageDict.wide_dict(message)),
+    )
+    content += "\n"
+    fence = get_unused_fence(content)
+    quoted_message = f"{fence}quote\n{message.content}\n{fence}"
+    content += quoted_message
+    return content
```

Comment:
This needs to handle truncating `message.content` if it's too long to fit the fence and quote and everything.

Reply:
Yeah, done.